ckanext/xloader/loader.py

```diff
@@ -42,15 +43,21 @@ class UnknownEncodingStream(object):
     only to run into problems later in the file.
     """
 
-    def __init__(self, filepath, file_format, **kwargs):
+    def __init__(self, filepath, file_format, decoding_result, **kwargs):
         self.filepath = filepath
         self.file_format = file_format
         self.stream_args = kwargs
+        self.decoding_result = decoding_result  # {'encoding': 'EUC-JP', 'confidence': 0.99}
 
     def __enter__(self):
         try:
-            self.stream = Stream(self.filepath, format=self.file_format,
-                                 **self.stream_args).__enter__()
+
+            if (self.decoding_result and self.decoding_result['confidence'] and self.decoding_result['confidence'] > 0.7):
```

Comment:
Could combine the second and third conditions with `get` eg `and self.decoding_result.get('confidence', 0) > 0.7`